### ISSUE TITLE
java.lang.NoClassDefFoundError: org/hamcrest/SelfDescribing

### ISSUE THREAD

**Issue by lbschenkel** [nebula-plugins/gradle-lint-plugin#31]
When using version 0.30.4 via
```groovy
plugins {
    id "java"
    id "nebula.lint" version "0.30.4"
}
```
and
```groovy
gradleLint.rules = ['all-dependency']
```
in my own project, I get the error below. I tried adding hamcrest dependency to the buildscript classpath but that didn't work. This happens on Gradle versions 2.8, 2.9, 2.13 and 2.14-rc-4.
```
:lintGradleError from [com.netflix.nebula.lint.rule.GradleLintRule$1] processing source file [null]
 FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':lintGradle'.
> org/hamcrest/SelfDescribing

* Try:
Run with --info or --debug option to get more log output.

* Exception is:
org.gradle.api.tasks.TaskExecutionException: Execution failed for task ':lintGradle'.
	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeActions(ExecuteActionsTaskExecuter.java:69)
	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.execute(ExecuteActionsTaskExecuter.java:46)
	at org.gradle.api.internal.tasks.execution.PostExecutionAnalysisTaskExecuter.execute(PostExecutionAnalysisTaskExecuter.java:35)
	at org.gradle.api.internal.tasks.execution.SkipUpToDateTaskExecuter.execute(SkipUpToDateTaskExecuter.java:66)
	at org.gradle.api.internal.tasks.execution.ValidatingTaskExecuter.execute(ValidatingTaskExecuter.java:58)
	at org.gradle.api.internal.tasks.execution.SkipEmptySourceFilesTaskExecuter.execute(SkipEmptySourceFilesTaskExecuter.java:52)
	at org.gradle.api.internal.tasks.execution.SkipTaskWithNoActionsExecuter.execute(SkipTaskWithNoActionsExecuter.java:52)
	at org.gradle.api.internal.tasks.execution.SkipOnlyIfTaskExecuter.execute(SkipOnlyIfTaskExecuter.java:53)
	at org.gradle.api.internal.tasks.execution.ExecuteAtMostOnceTaskExecuter.execute(ExecuteAtMostOnceTaskExecuter.java:43)
	at org.gradle.execution.taskgraph.DefaultTaskGraphExecuter$EventFiringTaskWorker.execute(DefaultTaskGraphExecuter.java:203)
	at org.gradle.execution.taskgraph.DefaultTaskGraphExecuter$EventFiringTaskWorker.execute(DefaultTaskGraphExecuter.java:185)
	at org.gradle.execution.taskgraph.AbstractTaskPlanExecutor$TaskExecutorWorker.processTask(AbstractTaskPlanExecutor.java:66)
	at org.gradle.execution.taskgraph.AbstractTaskPlanExecutor$TaskExecutorWorker.run(AbstractTaskPlanExecutor.java:50)
	at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor.process(DefaultTaskPlanExecutor.java:25)
	at org.gradle.execution.taskgraph.DefaultTaskGraphExecuter.execute(DefaultTaskGraphExecuter.java:110)
	at org.gradle.execution.SelectedTaskExecutionAction.execute(SelectedTaskExecutionAction.java:37)
	at org.gradle.execution.DefaultBuildExecuter.execute(DefaultBuildExecuter.java:37)
	at org.gradle.execution.DefaultBuildExecuter.access$000(DefaultBuildExecuter.java:23)
	at org.gradle.execution.DefaultBuildExecuter$1.proceed(DefaultBuildExecuter.java:43)
	at org.gradle.execution.DryRunBuildExecutionAction.execute(DryRunBuildExecutionAction.java:32)
	at org.gradle.execution.DefaultBuildExecuter.execute(DefaultBuildExecuter.java:37)
	at org.gradle.execution.DefaultBuildExecuter.execute(DefaultBuildExecuter.java:30)
	at org.gradle.initialization.DefaultGradleLauncher$4.run(DefaultGradleLauncher.java:153)
	at org.gradle.internal.Factories$1.create(Factories.java:22)
	at org.gradle.internal.progress.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:91)
	at org.gradle.internal.progress.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:53)
	at org.gradle.initialization.DefaultGradleLauncher.doBuildStages(DefaultGradleLauncher.java:150)
	at org.gradle.initialization.DefaultGradleLauncher.access$200(DefaultGradleLauncher.java:32)
	at org.gradle.initialization.DefaultGradleLauncher$1.create(DefaultGradleLauncher.java:98)
	at org.gradle.initialization.DefaultGradleLauncher$1.create(DefaultGradleLauncher.java:92)
	at org.gradle.internal.progress.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:91)
	at org.gradle.internal.progress.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:63)
	at org.gradle.initialization.DefaultGradleLauncher.doBuild(DefaultGradleLauncher.java:92)
	at org.gradle.initialization.DefaultGradleLauncher.run(DefaultGradleLauncher.java:83)
	at org.gradle.launcher.exec.InProcessBuildActionExecuter$DefaultBuildController.run(InProcessBuildActionExecuter.java:99)
	at org.gradle.tooling.internal.provider.ExecuteBuildActionRunner.run(ExecuteBuildActionRunner.java:28)
	at org.gradle.launcher.exec.ChainingBuildActionRunner.run(ChainingBuildActionRunner.java:35)
	at org.gradle.launcher.exec.InProcessBuildActionExecuter.execute(InProcessBuildActionExecuter.java:48)
	at org.gradle.launcher.exec.InProcessBuildActionExecuter.execute(InProcessBuildActionExecuter.java:30)
	at org.gradle.launcher.exec.ContinuousBuildActionExecuter.execute(ContinuousBuildActionExecuter.java:81)
	at org.gradle.launcher.exec.ContinuousBuildActionExecuter.execute(ContinuousBuildActionExecuter.java:46)
	at org.gradle.launcher.daemon.server.exec.ExecuteBuild.doBuild(ExecuteBuild.java:52)
	at org.gradle.launcher.daemon.server.exec.BuildCommandOnly.execute(BuildCommandOnly.java:36)
	at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:120)
	at org.gradle.launcher.daemon.server.exec.WatchForDisconnection.execute(WatchForDisconnection.java:37)
	at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:120)
	at org.gradle.launcher.daemon.server.exec.ResetDeprecationLogger.execute(ResetDeprecationLogger.java:26)
	at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:120)
	at org.gradle.launcher.daemon.server.exec.RequestStopIfSingleUsedDaemon.execute(RequestStopIfSingleUsedDaemon.java:34)
	at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:120)
	at org.gradle.launcher.daemon.server.exec.ForwardClientInput$2.call(ForwardClientInput.java:74)
	at org.gradle.launcher.daemon.server.exec.ForwardClientInput$2.call(ForwardClientInput.java:72)
	at org.gradle.util.Swapper.swap(Swapper.java:38)
	at org.gradle.launcher.daemon.server.exec.ForwardClientInput.execute(ForwardClientInput.java:72)
	at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:120)
	at org.gradle.launcher.daemon.server.health.DaemonHealthTracker.execute(DaemonHealthTracker.java:47)
	at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:120)
	at org.gradle.launcher.daemon.server.exec.LogToClient.doBuild(LogToClient.java:60)
	at org.gradle.launcher.daemon.server.exec.BuildCommandOnly.execute(BuildCommandOnly.java:36)
	at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:120)
	at org.gradle.launcher.daemon.server.exec.EstablishBuildEnvironment.doBuild(EstablishBuildEnvironment.java:72)
	at org.gradle.launcher.daemon.server.exec.BuildCommandOnly.execute(BuildCommandOnly.java:36)
	at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:120)
	at org.gradle.launcher.daemon.server.health.HintGCAfterBuild.execute(HintGCAfterBuild.java:41)
	at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:120)
	at org.gradle.launcher.daemon.server.exec.StartBuildOrRespondWithBusy$1.run(StartBuildOrRespondWithBusy.java:50)
	at org.gradle.launcher.daemon.server.DaemonStateCoordinator$1.run(DaemonStateCoordinator.java:237)
	at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:54)
	at org.gradle.internal.concurrent.StoppableExecutorImpl$1.run(StoppableExecutorImpl.java:40)
Caused by: java.lang.NoClassDefFoundError: org/hamcrest/SelfDescribing
	at com.netflix.nebula.lint.rule.dependency.ClassHierarchyUtils.typeHierarchy(ClassHierarchyUtils.groovy:32)
	at com.netflix.nebula.lint.rule.dependency.ClassHierarchyUtils$typeHierarchy.call(Unknown Source)
	at com.netflix.nebula.lint.rule.dependency.DependencyClassVisitor.visit(DependencyClassVisitor.groovy:88)
	at org.objectweb.asm.ClassReader.accept(Unknown Source)
	at org.objectweb.asm.ClassReader.accept(Unknown Source)
	at org.objectweb.asm.ClassReader$accept.call(Unknown Source)
	at com.netflix.nebula.lint.rule.dependency.DependencyService$2.visitFile(DependencyService.groovy:143)
	at com.netflix.nebula.lint.rule.dependency.DependencyService$2.visitFile(DependencyService.groovy)
	at java_nio_file_Files$walkFileTree.call(Unknown Source)
	at com.netflix.nebula.lint.rule.dependency.DependencyService.memoizedMethodPriv$classReferencesSourceSet(DependencyService.groovy:136)
	at com.netflix.nebula.lint.rule.dependency.DependencyService$_closure2.doCall(DependencyService.groovy)
	at com.netflix.nebula.lint.rule.dependency.DependencyService.classReferences(DependencyService.groovy)
	at com.netflix.nebula.lint.rule.dependency.DependencyService.memoizedMethodPriv$usedDependenciesString(DependencyService.groovy:240)
	at com.netflix.nebula.lint.rule.dependency.DependencyService$_closure6.doCall(DependencyService.groovy)
	at com.netflix.nebula.lint.rule.dependency.DependencyService.usedDependencies(DependencyService.groovy)
	at com.netflix.nebula.lint.rule.dependency.DependencyService$usedDependencies$7.call(Unknown Source)
	at com.netflix.nebula.lint.rule.dependency.UnusedDependencyRule$_visitGradleDependency_closure1.doCall(UnusedDependencyRule.groovy:52)
	at com.netflix.nebula.lint.rule.dependency.UnusedDependencyRule.visitGradleDependency(UnusedDependencyRule.groovy:51)
	at com.netflix.nebula.lint.rule.GradleLintRule.this$dist$invoke$4(GradleLintRule.groovy)
	at com.netflix.nebula.lint.rule.GradleLintRule$1.methodMissing(GradleLintRule.groovy)
	at com.netflix.nebula.lint.rule.GradleLintRule$1.this$dist$invoke$3(GradleLintRule.groovy)
	at com.netflix.nebula.lint.rule.GradleLintRule$1$2.methodMissing(GradleLintRule.groovy)
	at com.netflix.nebula.lint.rule.GradleLintRule$1$2.visitMethodCallInDependencies(GradleLintRule.groovy:384)
	at com.netflix.nebula.lint.rule.GradleLintRule$1$2.visitMethodCallExpression(GradleLintRule.groovy:207)
	at com.netflix.nebula.lint.rule.GradleLintRule$1$2.super$3$visitExpressionStatement(GradleLintRule.groovy)
	at com.netflix.nebula.lint.rule.GradleLintRule$1$2.visitExpressionStatement(GradleLintRule.groovy:344)
	at com.netflix.nebula.lint.rule.GradleLintRule$1$2.super$2$visitMethodCallExpression(GradleLintRule.groovy)
	at com.netflix.nebula.lint.rule.GradleLintRule$1$2.visitMethodCallExpression(GradleLintRule.groovy:214)
	at com.netflix.nebula.lint.rule.GradleLintRule$1$2.super$3$visitExpressionStatement(GradleLintRule.groovy)
	at com.netflix.nebula.lint.rule.GradleLintRule$1$2.visitExpressionStatement(GradleLintRule.groovy:344)
	at com.netflix.nebula.lint.rule.GradleLintRule$1$3$_visitConstructorOrMethod_closure11.doCall(GradleLintRule.groovy:464)
	at com.netflix.nebula.lint.rule.GradleLintRule$1$3.both(GradleLintRule.groovy:408)
	at com.netflix.nebula.lint.rule.GradleLintRule$1$3$both.callCurrent(Unknown Source)
	at com.netflix.nebula.lint.rule.GradleLintRule$1$3.visitConstructorOrMethod(GradleLintRule.groovy:464)
	at com.netflix.nebula.lint.org.codenarc.rule.AbstractAstVisitor.visitMethod(AbstractAstVisitor.java:171)
	at com.netflix.nebula.lint.org.codenarc.rule.AbstractAstVisitor.visitClass(AbstractAstVisitor.java:148)
	at com.netflix.nebula.lint.org.codenarc.rule.AbstractAstVisitorRule.applyTo(AbstractAstVisitorRule.java:97)
	at com.netflix.nebula.lint.org.codenarc.rule.AbstractRule.applyTo(AbstractRule.java:142)
	at com.netflix.nebula.lint.org.codenarc.rule.Rule$applyTo$0.call(Unknown Source)
	at com.netflix.nebula.lint.rule.GradleLintRule.applyTo(GradleLintRule.groovy:134)
	at com.netflix.nebula.lint.org.codenarc.rule.Rule$applyTo$0.call(Unknown Source)
	at com.netflix.nebula.lint.org.codenarc.rule.Rule$applyTo$0.call(Unknown Source)
	at com.netflix.nebula.lint.org.codenarc.analyzer.AbstractSourceAnalyzer.collectViolations(AbstractSourceAnalyzer.groovy:40)
	at com.netflix.nebula.lint.plugin.LintService$ReportableAnalyzer.analyze(LintService.groovy:48)
	at com.netflix.nebula.lint.plugin.LintService$ReportableAnalyzer$analyze.call(Unknown Source)
	at com.netflix.nebula.lint.plugin.LintService$_lint_closure3.doCall(LintService.groovy:96)
	at com.netflix.nebula.lint.plugin.LintService.lint(LintService.groovy:93)
	at com.netflix.nebula.lint.plugin.LintService$lint.call(Unknown Source)
	at com.netflix.nebula.lint.plugin.LintGradleTask.lint(LintGradleTask.groovy:41)
	at org.gradle.internal.reflect.JavaMethod.invoke(JavaMethod.java:75)
	at org.gradle.api.internal.project.taskfactory.AnnotationProcessingTaskFactory$StandardTaskAction.doExecute(AnnotationProcessingTaskFactory.java:228)
	at org.gradle.api.internal.project.taskfactory.AnnotationProcessingTaskFactory$StandardTaskAction.execute(AnnotationProcessingTaskFactory.java:221)
	at org.gradle.api.internal.project.taskfactory.AnnotationProcessingTaskFactory$StandardTaskAction.execute(AnnotationProcessingTaskFactory.java:210)
	at org.gradle.api.internal.AbstractTask$TaskActionWrapper.execute(AbstractTask.java:621)
	at org.gradle.api.internal.AbstractTask$TaskActionWrapper.execute(AbstractTask.java:604)
	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeAction(ExecuteActionsTaskExecuter.java:80)
	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeActions(ExecuteActionsTaskExecuter.java:61)
	... 68 more
Caused by: java.lang.ClassNotFoundException: org.hamcrest.SelfDescribing
	... 125 more


BUILD FAILED

Total time: 2.088 secs
```

**Comment by jkschneider** [comment]
Thanks @lbschenkel. Fixed and released with 0.30.5

**Comment by jouino16** [comment]
Hi @jkschneider, sorry if I'm asking the question in the wrong place.
Actually I'ma facing the same issue when applying the plugin (version 6.5.0) on my Android project.
I'am getting java.lang.NoClassDefFoundError: android/os/Bundle.
Do you have any idea how to get it fixed? Thank you in advance.
